app/src/main/java/com/lagradost/cloudstream3/ui/player/AbstractPlayerFragment.kt

```diff
@@ -607,23 +617,57 @@ abstract class AbstractPlayerFragment(
     }
 
     fun nextResize() {
-        resizeMode = (resizeMode + 1) % PlayerResize.entries.size
-        resize(resizeMode, true)
+        showAspectRatioDialog()
     }
 
     fun resize(resize: Int, showToast: Boolean) {
         resize(PlayerResize.entries[resize], showToast)
     }
 
+    private fun showAspectRatioDialog() {
+        player.handleEvent(CSPlayerEvent.Pause, PlayerEventSource.UI)
+
+        val aspectRatioOptions = PlayerResize.values().map { getString(it.nameRes) }
+
+        activity?.showDialog(
+            aspectRatioOptions,
+            -1,
```

Comment:
```suggestion
            DataStoreHelper.resizeMode,
```
Make the selected index what the user has previously picked